source/test/TestDirector.cc

```diff
@@ -4,21 +4,93 @@
 
 #include <abstract/File.hh>
 
+#include <egg/util/Stream.hh>
+
 #include <cstddef>
+#include <cstring>
 #include <format>
 
 namespace Test {
 
-TestDirector::TestDirector() {
+TestDirector::TestDirector(std::span<u8> &binaryData) : m_curTestIdx(0) {
+    Abstract::File::Remove("results.txt");
+    EGG::RamStream stream = EGG::RamStream(binaryData.data(), binaryData.size());
+    stream.setEndian(std::endian::big);
+    parseBinaryData(stream);
+    init();
+}
+
+TestDirector::~TestDirector() = default;
+
+void TestDirector::parseBinaryData(EGG::RamStream &stream) {
+    u16 numTestCases = stream.read_u16();
+    m_versionMajor = stream.read_u16();
+    m_versionMinor = stream.read_u16();
```

Comment:
Probably don't need to store these as members. Instead, maybe we have a `constexpr` major/minor versions in the test director, and we panic on version mismatch?

Reply:
In this case, `constexpr` works. To be clear, for KRKGs, we should ideally be able to support multiple versions, but since this suite is only(?) used in-house, this should be fine.